examples/solana/devnet/claim-stake/main.go

```diff
@@ -0,0 +1,187 @@
+ */
+
+func main() {
+	ctx := context.Background()
+
+	client, err := coinbase.NewClient(
+		coinbase.WithAPIKeyFromJSON(os.Args[1]),
+		coinbase.WithTimeout(20*time.Second),
+	)
+	if err != nil {
+		log.Fatalf("error creating coinbase client: %v", err)
+	}
+
+	address := coinbase.NewExternalAddress(string(networkID), os.Args[2])
+
+	balance, err := client.GetClaimableBalance(ctx, coinbase.Sol, address)
+	if err != nil {
+		log.Fatalf("error getting balance: %v", err)
+	}
+
+	log.Printf("Claimable balance: %s\n\n", balance.Amount().String())
+
+	stakingOperation, err := client.BuildClaimStakeOperation(ctx, amount, coinbase.Sol, address)
+	if err != nil {
+		log.Fatalf("error building claim stake operation: %v", err)
+	}
+
+	log.Printf("Staking operation ID: %s\n\n", stakingOperation.ID())
+
+	stakingOperation, err = client.Wait(ctx, stakingOperation, coinbase.WithWaitTimeoutSeconds(60))
+	if err != nil {
+		log.Fatalf("error waiting for staking operation: %v", err)
+	}
+
+	for _, transaction := range stakingOperation.Transactions() {
+		log.Printf("Unsigned tx payload: %s\n\n", transaction.UnsignedPayload())
+
+		signedTx, err := signSolTransaction(transaction.UnsignedPayload(), []string{os.Args[3]})
+		if err != nil {
+			log.Fatalf("error signing transaction: %v", err)
+		}
+
+		log.Printf("Signed tx: %s\n\n", signedTx)
+
+		broadcastedTx, err := broadcastSolTransaction(ctx, signedTx)
+		if err != nil {
+			log.Fatalf("error broadcasting transaction: %v", err)
+		}
+
+		log.Printf("Broadcasted tx: %s\n\n", getTxLink(stakingOperation.NetworkID(), broadcastedTx))
+	}
+}
+
+func signSolTransaction(unsignedTx string, privateKeys []string) (string, error) {
+	if len(privateKeys) == 0 {
+		return "", fmt.Errorf("need to pass at least one private key")
+	}
+
+	signers := make([]solana.PrivateKey, 0, len(privateKeys))
+
+	for _, privateKey := range privateKeys {
+		signer, err := solana.PrivateKeyFromBase58(privateKey)
+		if err != nil {
+			return "", fmt.Errorf("error getting private key: %w", err)
+		}
+
+		signers = append(signers, signer)
+	}
+
+	data := base58.Decode(unsignedTx)
+
+	// parse transaction
+	tx, err := solana.TransactionFromDecoder(bin.NewBinDecoder(data))
+	if err != nil {
+		return "", err
+	}
+
+	// clear signatures: https://github.com/gagliardetto/solana-go/issues/186
+	tx.Signatures = nil
+
+	if _, err = tx.Sign(func(key solana.PublicKey) *solana.PrivateKey {
+		for _, candidate := range signers {
+			if candidate.PublicKey().Equals(key) {
+				return &candidate
+			}
+		}
+
+		return nil
+	}); err != nil {
+		return "", fmt.Errorf("error signing transaction: %w", err)
+	}
+
+	marshaledTx, err := tx.MarshalBinary()
+	if err != nil {
+		return "", fmt.Errorf("error marshaling transaction: %w", err)
+	}
+
+	base58EncodedSignedTx := base58.Encode(marshaledTx)
+
+	return base58EncodedSignedTx, nil
+}
+
+func broadcastSolTransaction(ctx context.Context, signedTx string) (string, error) {
```

Comment:
I really think we should not be duplicating the sign/broadcast for every variation.

Reply:
I agree. Let me see if we can incorporate this into the SDK to help standardize this functionality.